components/team_controller/team_controller.tsx

```diff
@@ -0,0 +1,258 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {lazy, memo, useEffect, useRef, useState} from 'react';
+import {Route, Switch, useHistory, useParams} from 'react-router-dom';
+import iNoBounce from 'inobounce';
+
+import {Team} from '@mattermost/types/teams';
+import {ServerError} from '@mattermost/types/errors';
+
+import {ActionResult} from 'mattermost-redux/types/actions';
+
+import {reconnect} from 'actions/websocket_actions.jsx';
+import {emitCloseRightHandSide} from 'actions/global_actions';
+
+import Constants from 'utils/constants';
+import {isIosSafari} from 'utils/user_agent';
+import {cmdOrCtrlPressed, isKeyPressed} from 'utils/utils';
+
+import {makeAsyncComponent} from 'components/async_load';
+import ChannelController from 'components/channel_layout/channel_controller';
+
+import LocalStorageStore from 'stores/local_storage_store';
+
+import type {OwnProps, PropsFromRedux} from './index';
+
+const BackstageController = makeAsyncComponent('BackstageController', lazy(() => import('components/backstage')));
+const Pluggable = makeAsyncComponent('Pluggable', lazy(() => import('plugins/pluggable')));
+
+const WAKEUP_CHECK_INTERVAL = 30000; // 30 seconds
+const WAKEUP_THRESHOLD = 60000; // 60 seconds
+const UNREAD_CHECK_TIME_MILLISECONDS = 10000;
+
+declare global {
+    interface Window {
+        isActive: boolean;
+    }
+}
+
+type Props = PropsFromRedux & OwnProps;
+
+function TeamController(props: Props) {
+    const history = useHistory();
+    const {team: teamNameParam} = useParams<Props['match']['params']>();
+
+    const [team, setTeam] = useState<Team | null>(getTeamFromTeamList(props.teamsList, teamNameParam));
+
+    const blurTime = useRef(new Date().getTime());
+    const lastTime = useRef(Date.now());
+
+    function handleFocus() {
+        if (props.selectedThreadId) {
+            window.isActive = true;
+        }
+        if (props.currentChannelId) {
+            window.isActive = true;
+            props.markChannelAsReadOnFocus(props.currentChannelId);
+        }
+
+        const currentTime = (new Date()).getTime();
+        if ((currentTime - blurTime.current) > UNREAD_CHECK_TIME_MILLISECONDS && props.currentTeamId) {
```

Comment:
Probably this logic is brought from a different place, but seems wrong.

Are we fetching all channels and members every time we have the window not focused for more than 10 seconds? Wouldn't the websocket take care of things to be up to date without this?

Reply:
I think this logic might actually date back to before we trusted that all server would even have a websocket connection. That's probably something we should reconsider now that we rely on the websocket and even have all that "reliable websockets" stuff, but I'd leave it for a later change